rules/integrations/google_workspace/persistence_google_workspace_role_modified.toml

```
@@ -24,7 +24,55 @@ interval = "10m"
 language = "kuery"
 license = "Elastic License v2"
 name = "Google Workspace Role Modified"
-note = """## Setup
+note = """## Triage and analysis
+
+### Investigating Google Workspace Role Modified
+
+Google Workspace roles allow administrators to assign specific permissions to users or groups where the principle of least privilege (PoLP) is recommended. Admin roles in Google Workspace grant users access to the Google Admin console, where further domain-wide settings are accessible. Google Workspace contains prebuilt admin roles for performing business functions related to users, groups, and services. Custom administrator roles can be created where prebuilt roles are not preferred. Each Google Workspace service has a set of custodial privileges that can be added to custom roles.
+
+Roles assigned to users will grant them additional permissions and privileges within the Google Workspace domain. Threat actors may modify existing roles with new privileges to thus further their intrusion efforts and laterally move throughout the organization. Users with unexpected privileges may also cause operational disfunction if unfamiliar settings are adjusted without warning.
+
+This rule identifies when a Google Workspace role is modified.
+
+#### Possible investigation steps
+
+- Identify the user account(s) associated by reviewing `user.name` or `user.email` in the alert.
+- Identify the role modified by reviewing `google_workspace.admin.role.name` in the alert.
+- Identify the privilege added or removed by reviewing `google_workspace.admin.privilege.name` in the alert.
+- With the user identified, verify administrative privileges are scoped properly.
+- To identify other users with this role, search for `event.action: ASSIGN_ROLE`
+  - Add `google_workspace.admin.role.name` with the role added as an additional filter.
+  - Adjust the relative time accordingly to identify all users that were assigned this role.
+- Identify if the user account was recently created by searching for `event.action: CREATE_USER`.
+- If a privilege was added, monitor users assigned this role for the next 24 hours in an attempt to identify using the new privilege.
```

Possible revision, ignore if not correct: 

```suggestion
- If a privilege was added, monitor users assigned this role for the next 24 hours and look for attempts to use the new privilege.
```